src/Config.cpp

```diff
@@ -243,128 +210,99 @@ bool	Config::pathExists(std::map<std::string, std::string> list, std::string var
-			start = i;
-			while (i < it->second.length() && it->second[i] != ' ')
-				i++;
-			if (access((it->second.substr(start, i - start)).c_str(), F_OK) == 0){
-				if (isVulnerablePath(it->second.substr(start, i - start)))
-					return (error("Config file: Path is vulnerable"));
-				count1++;
-				continue ;
-			}
-			if (!isValidErrorCode(it->second.substr(start, i - start)))
-				return (error("Config file: Invalid error_page"));
-			count2++;
-		}
-		if (count1 == 0 || count1 > 1 || count2 == 0)
-			return (error("Config file: Invalid error_page"));
-	}
-	return (true);
-}
-
-
-bool	Config::checkVariablesValue(std::map<std::string, std::string> var)
-{
-	std::string tmp_meth[] = {"GET", "POST", "DELETE"};
-	std::string tmp_cgi[] = {".py", ".php", ".pl", ".cgi"};
-	std::list<std::string> methods(tmp_meth, tmp_meth + sizeof(tmp_meth) / sizeof(tmp_meth[0]));
-	std::list<std::string> cgi_ext(tmp_cgi, tmp_cgi + sizeof(tmp_cgi) / sizeof(tmp_cgi[0]));
-	std::map<std::string, std::string>::iterator it;
-	unsigned int	start = 0;
-
-	// // ROOT
-	// if (!pathExists(var, "root"))
-	// 	return (false);
-	// ALIAS
-	if (!pathExists(var, "alias"))
-		return (false);
-	// CGI_PATH
-	if (!pathExists(var, "cgi_path"))
-		return (false);
-	// ERROR_PAGE
-	if (!checkErrorPage(var))
-		return (false);
-	// ALLOW_METHODS
-	it = var.find("allow_methods");
-	if (it != var.end()){
-		for (unsigned int i = 0; i < it->second.length(); ++i){
-			start = i;
-			while (i < it->second.length() && it->second[i] != ' ')
-				i++;
-			if (std::find(methods.begin(), methods.end(), it->second.substr(start, i - start)) == methods.end())
-				return (error("Config file: Invalid allow_method"));
-		}
-	}
-	// AUTOINDEX
-	it = var.find("autoindex");
-	if (it != var.end())
-		if (it->second != "on")
-			return (error("Config file: Invalid autoindex"));
-	// CLIENT_MAX_BODY_SIZE
-	it = var.find("client_max_body_size");
-	if (it != var.end()){
-		if (!isNumber(it->second))
-			return (error("Config file: Invalid client_max_body_size"));
-	}
-	// CGI_EXT
-	it = var.find("cgi_ext");
-	if (it != var.end()){
-		for (unsigned int i = 0; i < it->second.length(); ++i){
-			start = i;
-			while (i < it->second.length() && it->second[i] != ' ')
-				i++;
-			if (std::find(cgi_ext.begin(), cgi_ext.end(), it->second.substr(start, i - start)) == cgi_ext.end())
-				return (error("Config file: Invalid cgi_ext"));
-		}
-	}
-	// INDEX
-	it = var.find("index");
-	if (it != var.end()){
-		for (unsigned int i = 0; i < it->second.length(); ++i){
-			start = i;
-			while (i < it->second.length() && it->second[i] != ' ')
-				i++;
-			if (access((it->second.substr(start, i - start)).c_str(), F_OK) == 0){
-				if (isVulnerablePath(it->second.substr(start, i - start)))
-					return (error("Config file: Path is vulnerable"));
-				return (true);
-			}
-		}
-		return (error("Config file: Invalid index"));
-	}
-	return (true);
-}
+// bool	Config::checkVariablesValue(Variables var)
```

Comment:
Why so much uncommented code? Do we still need it?

Reply:
I might need it in the future